File: Graph/isConnected/isConnected.js
(function () {
    
    function DFS(edges, n, sv, visited) {
        visited[sv] = true;

        for (let i=0; i<n; i++) {
            if (i === sv) continue;

            if (edges[sv][i] && !visited[i]) {
                DFS(edges,n,i,visited);
            }
        }
    }   

    function isConnected(edges, n) {
        let visited = [];
        for (let i = 0; i<n; i++) visited[i] = false;

        DFS(edges,n,0,visited);

        for (let i of visited) {
            if (!i) return false;
        }

        return true;
    }

    (function main(){

        let n = 4;
        let e = 4;

        // let n = 6;
        // let e = 3;
        
        let connection = [
            {sv:0, ev:1},
            {sv:0, ev:3},
            {sv:1, ev:2},
            {sv:2, ev:3},
        ];

        // let connection = [
        //     {sv:5, ev:3},
        //     {sv:0, ev:1},
        //     {sv:3, ev:4},
        // ];


        let edges = [];
        for (let i = 0; i<n; i++) {
            edges[i] = [];
            for (let j = 0; j<n; j++) {
                edges[i][j] = false;
            }
        }

        for (let i = 0; i<e; i++) {
            let {sv,ev} = connection[i];
            edges[sv][ev] = true;
            edges[ev][sv] = true;
        }


        let ans = isConnected(edges,n);
        console.log(ans);
    }());
}());

File: Graph/isConnected/isConnected.cpp
#include<iostream>
using std::cin;
using std::cout;
using std::endl;
using std::boolalpha;

void DFS(bool** edges, int n, bool* visited, int sv) {
    visited[sv] = true;
    for (int i{}; i < n; i++) {
        if (i == sv) continue;
        if (edges[sv][i] && !visited[i]) DFS(edges, n, visited, i);
    }
}

bool isConnected(bool** edges, int n) {
    bool* visited{new bool[n]};
    for(int i{}; i < n; i++) visited[i] = false;
    DFS(edges, n, visited, 0);
    for (int i{}; i < n; i++) if (!visited[i]) return false;
    delete [] visited;
    return true;
}

int main() {
    int n, e;
    cin >> n >> e;

    bool** edges{new bool*[n]};
    for (int i{}; i < n; i++) {
        edges[i] = new bool[n];
        for (int j{}; j < n; j++) edges[i][j] = false;
    }

    for(int i{}; i < e; i++) {
        int sv, ev;
        cin >> sv >> ev;
        edges[sv][ev] = true;
        edges[ev][sv] = true;
    }

    cout << boolalpha << isConnected(edges, n) << endl;

    for(int i{}; i < n; i++) delete [] edges[i];
    delete [] edges;
}
